### PR TITLE
Added readiness probe script

### PR DESCRIPTION
**Pull request by lucferbux**
## Related Issues and Dependencies

https://issues.redhat.com/browse/ODH-453

## This introduces a breaking change

- [ ] Yes
- [X] No

Adds a new file, which can act as an livenessProbe script

## This Pull Request implements

Add an script that works as a liveness probe for jupyterhub on odh.

### CODE REVIEW

**Comment by lucferbux** [comment]
Ok, seems to be working, i've left my cluster running with those changes in order to check if the issue QA has found appear there too.

File: readinessProbe.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+ME="$(cat /etc/hostname)"
+
+function is_running() {
```
Comment (vpavlin) [inline readinessProbe.sh:3]: Can you make the function names more descriptive? What `is_running`? What `is_ready`?

File: readinessProbe.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+function is_ready() {
+    [ $(curl -s -o /dev/null -w ''%{http_code}'' localhost:4040) == "200" ]
+}
+
+function get_leader() {
+    echo "$(curl http://localhost:4040 2> /dev/null | python3 -c "import sys, json; print(json.load(sys.stdin)['name'])")"
+}
+
+function get_pod_name() {
+    echo ${ME}
```
Comment (vpavlin) [inline readinessProbe.sh:16]: Might make sense to use `${HOSTNAME}` to avoid creating a custom env var for it

File: readinessProbe.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+if ! is_running; then
+    # Pod is waiting to become leader
+    exit 0
+elif is_leader; then
+    # Pod is the leader
+    exit 0
+else
+    # Pod is still running but it is not the leader
+    exit 1
+fi
```
Comment (vpavlin) [inline readinessProbe.sh:37]: missing new line

Comment (lucferbux) [reply readinessProbe.sh]: 👍

File: readinessProbe.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
+ME="$(cat /etc/hostname)"
```
[the remark this answers is not in the document]
Comment (lucferbux) [reply readinessProbe.sh]: 👍

**Comment by vpavlin** [comment]
lgtm

**Pull request merged**